src/include/compat.h

```diff
@@ -13,10 +13,57 @@
 #define CEPH_COMPAT_H
 
 #if defined(__FreeBSD__)
+
+/* Make sure that ENODATA is defined in the correct way */
+#ifndef ENODATA
 #define	ENODATA	ENOATTR
+#else
+#if (ENODATA == 9919)
+// #warning ENODATA already defined to be 9919, redefining to fix
```

Comment:
On 18-5-2016 22:19, Sage Weil wrote:
> In src/include/compat.h
> <https://github.com/ceph/ceph/pull/9178#discussion_r63774437>:
> 
>>  #define	ENODATA	ENOATTR
>> +#else
>> +#if (ENODATA == 9919)
>> +// #warning ENODATA already defined to be 9919, redefining to fix
> 
> why is this commented out?

I had it active in first, but it blasts so much warnings in a lot of
source files where the boost code is used as well. And it is sort of a
warning overflow (death by warnings :) ). I think it deserves to be
turned on when all references to compat.h are moved to the start of the
code files.

And then it deserves a rewrite to make it a compile check that is done
inline at the location of reference, not generating runtime overhead. I
have to admit that I've never done such a thing.

But it is there to remind the actual origin of the trouble.

--WjW